File: app/src/main/java/com/orozco/netreport/inject/ApplicationComponent.java
package com.orozco.netreport.inject;

import dagger.Component;

@PerApplication
@Component(modules = {
        ApplicationModule.class,
        RestModule.class,
        ActionCreatorModule.class,
        ModelModule.class,
        StoreModule.class
})
public interface ApplicationComponent {
    ActivityComponent plus(ActivityModule activityModule);
}

File: app/src/main/java/com/orozco/netreport/inject/ApplicationComponent.kt
package com.orozco.netreport.inject

import com.orozco.netreport.BASS
import dagger.Component

@PerApplication
@Component(modules = arrayOf(ApplicationModule::class, RestModule::class, ActionCreatorModule::class, ModelModule::class, StoreModule::class))
interface ApplicationComponent {
    fun plus(activityModule: ActivityModule): ActivityComponent
    fun inject(app: BASS)

}
